lib/datasets/aozora-bunko.rb

```diff
@@ -0,0 +1,210 @@
+
+    class Record
+      attr_writer :cache_path
+
+      def initialize(*args)
+        super
+        @text = nil
+        @html = nil
+        @cache_path = nil
+      end
+
+      def text
+        return @text unless @text.nil?
+        return @text if text_file_url.nil? || text_file_url.empty?
+
+        # when url is not zip file, it needs to open web page by brower and has to download
+        # e.g. https://mega.nz/file/6tMxgAjZ#PglDDyJL0syRhnULqK0qhTMC7cktsgqwObj5fY_knpE
+        return @text unless text_file_url.end_with?('.zip')
+
+        downloader = Downloader.new(text_file_url)
+        downloader.download(text_file_output_path)
+
+        @text = ZipExtractor.new(text_file_output_path).extract_one_file do |input|
+          input.read.encode(Encoding::UTF_8, normalize_encoding(text_file_character_encoding))
+        end
+
+        @text
+      end
+
+      def html
+        return @html unless @html.nil?
+        return @html if html_file_url.nil? || html_file_url.empty?
+
+        downloader = Downloader.new(html_file_url)
+        downloader.download(html_file_output_path)
+        @html = File.read(html_file_output_path).encode(Encoding::UTF_8,
+                                                        normalize_encoding(html_file_character_encoding))
+
+        @html
+      end
+
+      private
+
+      def text_file_output_path
+        cache_base_dir + text_file_name
+      end
+
+      def html_file_output_path
+        cache_base_dir + html_file_name
+      end
+
+      def text_file_name
+        text_file_url.split('/').last
+      end
+
+      def html_file_name
+        html_file_url.split('/').last
+      end
+
+      def cache_base_dir
+        @cache_path.base_dir + title_id + person_id
+      end
+
+      def normalize_encoding(encoding)
+        case encoding
+        when 'ShiftJIS'
+          Encoding::Shift_JIS
+        when 'EUC'
+          # EUC is setted at only :html_file_character_encoding
+          # :html_file_character_encoding value is EUC but the actual condition of encoding is UTF_8 in all case.
+          # e.g. html_file_url is 'http://literature.hanagasumi.net/YoungGoodmanBrownJP.html'
+          #      html_file_character_encoding is 'EUC'
+          #      the actual condition of encoding is UTF_8
+          Encoding::UTF_8
+        when 'UTF-8'
+          Encoding::UTF_8
+        else
+          encoding
+        end
+      end
+    end
+
+    def initialize
+      super()
+
+      @metadata.id = 'aozora-bunko'
+      @metadata.name = 'Aozora Bunko'
+      @metadata.url = 'https://www.aozora.gr.jp/'
+      @metadata.licenses = 'CC-BY-2.1-JP'
+      @metadata.description = <<~DESCRIPTION
+        Aozora Bunko is an activity to collect free electronic books that anyone can access
+        on the Internet like a library. The copyrighted works and the works that are said to be
+        "free to read" are available after being digitized in text and XHTML (some HTML) formats.
+      DESCRIPTION
```

Comment:
Should I quote from wikipedia?
https://en.wikipedia.org/wiki/Aozora_Bunko

Reply:
Many of the works are [public domain](https://en.wikipedia.org/wiki/Public_domain), though some have been donated as free to read.  If there is a distinction between public domain and free to read, it may be helpful to have  such a field - it is unclear if copyrighted covers this. As an example, many [Creative Commons licenses](https://creativecommons.org/) are free to read, but not public domain since they typically require one to acknowledge the creator of the original material when reusing the material.

Reply:
The current description translates the description at https://www.aozora.gr.jp/guide/aozora_bunko_hayawakari.html , right?

> 　青空文庫は、誰にでもアクセスできる自由な電子本を、図書館のようにインターネット上に集めようとする活動です。
>　著作権の消滅した作品と、「自由に読んでもらってかまわない」とされたものを、テキストとＸＨＴＭＬ（一部はＨＴＭＬ）形式に電子化した上で揃えています。

I think that it's OK with the current description because it's the official description.


The book list, that is parsed in this class, is licensed under CC-BY-2.1-JP:

https://www.aozora.gr.jp/guide/kijyunn.html

> 　あなたは、図書カードや書架情報（作家別作品一覧CSVファイル）のデータを、ダウンロードして利用することができます。
>　データは、有償・無償であるかを問わず、データベースや読書ソフトウェアにおける書誌情報として、活用できます。
>　こうしたデータは、「クリエイティブ・コモンズ 表示 2.1 日本 ライセンス」のもとで提供いたします。

Copyright of each book can be confirmed by `record.copyrighted?`.